core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -617,14 +610,14 @@ public synchronized CatchupFuture catchup(Map<String, Long> journalSequenceNumbe
 
   @Override
   public synchronized void checkpoint() throws IOException {
-    // TODO(feng): consider removing this once we can automatically propagate
-    //             snapshots from standby master
     try (RaftJournalAppender client = new RaftJournalAppender(mServer, this::createClient,
-        mRawClientId)) {
-      mSnapshotAllowed.set(true);
+        mRawClientId); RaftClient raftClient = createClient()) {
```

Comment:
are we sure this client is always going to connect to the same machine this function is being called from?

Reply:
By default, the `RaftClient#getSnapshotManagementApi#create` goes to the leader ([link](https://github.com/apache/ratis/blob/56a8e1730059fdc4e50544e85b09318851f1a492/ratis-client/src/main/java/org/apache/ratis/client/impl/SnapshotManagementImpl.java#L35-L41)).
I'll change it to make it go explicitly to the same machine this function is being called from.